sycl/include/CL/sycl/detail/defines_elementary.hpp

```diff
@@ -14,6 +14,55 @@
 #define __SYCL_INLINE_NAMESPACE(X) namespace X
 #endif // __SYCL_DISABLE_NAMESPACE_INLINE__
 
+// TODO: Disable macro should be removed once transition to sycl:: is done.
+#ifndef __SYCL_DISABLE_SYCL121_NAMESPACE
+#define __SYCL_ENABLE_SYCL121_NAMESPACE
+#endif
+
+#ifdef __SYCL_ENABLE_SYCL121_NAMESPACE
+// Old SYCL1.2.1 namespace scheme
+#define __SYCL_NS_OPEN_1 __SYCL_INLINE_NAMESPACE(cl)
```

Comment:
Can we remove `__SYCL_INLINE_NAMESPACE`? I think it's not supposed to be used explicitly anymore. Right?

Reply:
It is still used for builtins declaration - https://github.com/intel/llvm/blob/sycl/sycl/include/CL/sycl/detail/builtins.hpp#L106 and will be used until we remove support for old namespaces.

Reply:
Will get rid of this macro in the second patch(by introducing a new one).

Reply:
> It is still used for builtins declaration - https://github.com/intel/llvm/blob/sycl/sycl/include/CL/sycl/detail/builtins.hpp#L106 and will be used until we remove support for old namespaces.

```c++
#ifndef __SYCL_DEVICE_ONLY__
__SYCL_INLINE_NAMESPACE(cl) {
namespace __host_std {
```

I think we should re-visit the design of this code.
It is unclear to me why we put built-in declarations to `cl` namespace. Another question: why do we inline `cl` namespace here? I understand that it's needed to allow users to use `sycl` namespace w/o `cl::` qualifier, but `__host_std` namespace should not be exposed like this. Right?

> Will get rid of this macro in the second patch(by introducing a new one).

Could you clarify what new macro is for?

If this is the only use, can we just use `inline namespace` instead of the macro here? I'm okay to make this change in follow-up patches.